modules/working-file-repository-service-impl/src/main/java/org/opencastproject/workingfilerepository/impl/WorkingFileRepositoryImpl.java

```diff
@@ -208,6 +225,29 @@ public void deactivate() {
     }
   }
 
+  /**
+   * Returns the filename translated into a version that can safely be used as part of a file system path.
+   *
+   * The method shortens both the base file name and the extension to a maximum of 255 characters each,
+   * and replaces unsafe characters with <doce>_</doce>.
+   *
+   * @param fileName
+   *          The file name
+   * @return the safe version
+   */
+  @Override
+  public String toSafeName(String fileName) {
+    var extension = FilenameUtils.getExtension(fileName)
+        .replaceAll(filenameRegex, "_");
+    var baseName = FilenameUtils.getBaseName(fileName)
```

Comment:
This doesn't appear to do what's documented in the .properties configuration file (i.e. the baseName does not contain the path part):

```
#   original:  -problematic-file/../name&&&%.ext
#   converted: _problematic-file____name____.ext
```

which is okay but documentation might need an update.